actors/market/src/lib.rs

```diff
@@ -1025,6 +979,81 @@ impl Actor {
             }
         }
     }
+
+    fn process_deal_updates(
+        rt: &impl Runtime,
+        params: ProcessDealsParams,
+    ) -> Result<ProcessDealsReturn, ActorError> {
+        let curr_epoch = rt.curr_epoch();
+        let mut total_slashed = TokenAmount::zero();
+        let mut terminated_deals: Vec<DealID> = Vec::new();
+
+        rt.transaction(|st: &mut State, rt| {
+            for deal_id in params.deal_ids {
+                let deal_proposal = st.get_proposal(rt.store(), deal_id)?;
+                let dcid = rt_deal_cid(rt, &deal_proposal)?;
+                let (deal_state, slashed) = st.get_active_deal_or_cleanup(
+                    rt.store(),
+                    curr_epoch,
+                    deal_id,
+                    &deal_proposal,
+                    &dcid,
+                )?;
+
+                let mut deal_state = match deal_state {
+                    Some(deal_state) => deal_state,
+                    None => {
+                        // deal was cleaned up
+                        total_slashed += slashed;
+                        terminated_deals.push(deal_id);
+                        continue;
+                    }
+                };
+
+                let (slash_amount, remove_deal) =
+                    st.process_deal_update(rt.store(), &deal_state, &deal_proposal, curr_epoch)?;
+
+                if slash_amount.is_negative() {
+                    return Err(actor_error!(
+                        illegal_state,
+                        format!(
+                            "computed negative slash amount {} for deal {}",
+                            slash_amount, deal_id
+                        )
+                    ));
+                }
+
+                if remove_deal {
+                    total_slashed += slash_amount;
+                    st.remove_completed_deal(rt.store(), deal_id)?;
+                    terminated_deals.push(deal_id);
+                } else {
+                    if !slash_amount.is_zero() {
+                        return Err(actor_error!(
+                            illegal_state,
+                            "continuing deal {} should not be slashed",
+                            deal_id
+                        ));
+                    }
+
+                    deal_state.last_updated_epoch = curr_epoch;
+                    st.put_deal_states(rt.store(), &[(deal_id, deal_state)])?;
```

Comment:
Since we're writing new code here, can you pull the write out of the loop, and batch put all the states?